exterminate.lua

```diff
@@ -25,11 +25,21 @@ local function isUnitFriendly(unit)
     if dfhack.units.isDanger(unit) then
         return false
     end
+    local isInAdvGroup = false
+    local isAdvPet = false
+    local adv = dfhack.world.getAdventurer()
+    if adv then
+        isInAdvGroup = unit.relationship_ids.GroupLeader == dfhack.world.getAdventurer().id
+        isAdvPet = unit.relationship_ids.PetOwner == dfhack.world.getAdventurer().id
+    end
     return dfhack.units.isOwnCiv(unit) or
         dfhack.units.isOwnGroup(unit) or
```

Comment:
how does isOwnGroup behave in adventure mode?

Reply:
it doesn't. isOwnGroup only checks plotinfo which is irrelevant to adventure mode

Reply:
does that mean it always returns false, or that it returns garbage data?